pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -123,6 +123,11 @@ func (cd *clusterDeploy) doSync(cluster *v3.Cluster) error {
 		return nil
 	}
 
+	_, err := cd.systemAccountManager.GetOrCreateSystemClusterToken(cluster.Name)
+	if err != nil {
+		return err
+	}
```

Comment:
Thanks for bring this up because I was confused about it, see the commit message:

```
Create system cluster token on first sync attempt

If a cluster provisioner has deployed a remote cluster with a private
control plane endpoint, it will need to have the cluster agent deployed
manually. Without this patch, it will not be able to serve the agent
YAML manifest yet because doSync exits early since it cannot find any
nodes. In order to be able to generate the agent manifest URL, this
change goes ahead and generates the cluster registration token as soon
as the cluster is deployed.
```

In short, when deploying a private cluster in GKE I couldn't do the manual agent install step without this. I don't know why it's a problem here and not a problem for EKS, do you?

Reply:
@thedadams can you review this, and also the last two commits in this PR for private cluster approach.

Reply:
The way this works in the UI now is the following:
1. A user creates a private cluster.
2. Once the cluster is running, the UI gives the user the option to see the import command.
3. If the user clicks to see the import command, the UI makes a GET request to `/v3/clusterregistrationtokens` to see if the cluster has any registration tokens.
3a. If the cluster does not have any registration tokens, the UI makes a POST request to `/v3/clusterregistrationtokens` to create a token and the repeats step (3).

You are essentially doing (3) and (3a) automatically for every cluster that is created in Rancher. This is not the correct flow because not every cluster needs a registration token.

I believe removing this code and letting the UI handle this as above is the correct way to do it. I'm happy to help if you would like to test this with your development environment.